From: Tomas Henzl <thenzl@redhat.com>
Date: Wed, 21 Oct 2009 13:28:28 -0400
Subject: [scsi] mpt2sas: use selected regions
Message-id: <4ADF0C7C.4050702@redhat.com>
Patchwork-id: 21176
O-Subject: [RHEL5.5 PATCH 2/2] mpt2sas: use selected regions
Bugzilla: 516702

bz#516702

Make use of pci_request_selected_regions instead of
pci_request_regions. This is also upstream.

Signed-off-by: Don Zickus <dzickus@redhat.com>

diff --git a/drivers/scsi/mpt2sas/mpt2sas_base.c b/drivers/scsi/mpt2sas/mpt2sas_base.c
index 1c49a8c..1a7cebd 100644
--- a/drivers/scsi/mpt2sas/mpt2sas_base.c
+++ b/drivers/scsi/mpt2sas/mpt2sas_base.c
@@ -1121,15 +1121,18 @@ mpt2sas_base_map_resources(struct MPT2SAS_ADAPTER *ioc)
 	dinitprintk(ioc, printk(MPT2SAS_DEBUG_FMT "%s\n",
 	    ioc->name, __func__));
 
-	if (pci_enable_device(pdev)) {
-		printk(MPT2SAS_WARN_FMT "pci_enable_device: failed\n",
-		    ioc->name);
+	ioc->bars = pci_select_bars(pdev, IORESOURCE_MEM);
+	if (pci_enable_device_mem(pdev)) {
+		printk(MPT2SAS_WARN_FMT "pci_enable_device_mem: "
+		    "failed\n", ioc->name);
 		return -ENODEV;
 	}
 
-	if (pci_request_regions(pdev, MPT2SAS_DRIVER_NAME)) {
-		printk(MPT2SAS_WARN_FMT "pci_request_regions: failed\n",
-		    ioc->name);
+
+	if (pci_request_selected_regions(pdev, ioc->bars,
+	    MPT2SAS_DRIVER_NAME)) {
+		printk(MPT2SAS_WARN_FMT "pci_request_selected_regions: "
+		    "failed\n", ioc->name);
 		r = -ENODEV;
 		goto out_fail;
 	}
@@ -1184,7 +1187,7 @@ mpt2sas_base_map_resources(struct MPT2SAS_ADAPTER *ioc)
 		iounmap(ioc->chip);
 	ioc->chip_phys = 0;
 	ioc->pci_irq = -1;
-	pci_release_regions(pdev);
+	pci_release_selected_regions(ioc->pdev, ioc->bars);
 	pci_disable_device(pdev);
 	return r;
 }
@@ -3245,7 +3248,7 @@ mpt2sas_base_free_resources(struct MPT2SAS_ADAPTER *ioc)
 		iounmap(ioc->chip);
 	ioc->pci_irq = -1;
 	ioc->chip_phys = 0;
-	pci_release_regions(pdev);
+	pci_release_selected_regions(ioc->pdev, ioc->bars);
 	pci_disable_device(pdev);
 	return;
 }
diff --git a/drivers/scsi/mpt2sas/mpt2sas_base.h b/drivers/scsi/mpt2sas/mpt2sas_base.h
index e176f75..1076ab2 100644
--- a/drivers/scsi/mpt2sas/mpt2sas_base.h
+++ b/drivers/scsi/mpt2sas/mpt2sas_base.h
@@ -590,6 +590,7 @@ struct MPT2SAS_ADAPTER {
 	unsigned long	pio_chip;
 	int		logging_level;
 	u8		ir_firmware;
+	int		bars;
 	u8		mask_interrupts;
 
 	/* fw fault handler */
